src/modules/idyntree-yarp-visualizer/RobotConnectors.cpp

```diff
@@ -0,0 +1,558 @@
+#include <yarp/os/LogStream.h>
+#include <iDynTree/Core/Utils.h>
+#include <yarp/dev/IAxisInfo.h>
+#include "RobotConnectors.h"
+#include <algorithm>
+
+YARP_DECLARE_PLUGINS(ReadOnlyRemoteControlBoardLib);
+
+namespace idyntree_yarp_tools {
+
+bool RemapperConnector::getOrGuessControlBoardsFromFile(const yarp::os::Searchable &inputConf)
+{
+    m_controlBoards.clear();
+    yarp::os::Value controlBoardsValue = inputConf.find("controlboards");
+
+    if (controlBoardsValue.isNull())
+    {
+        m_controlBoards = {"head", "torso", "left_arm", "right_arm", "left_leg", "right_leg"}; //assumes these as controlboards
+    }
+    else
+    {
+        if (!controlBoardsValue.isList())
+        {
+            yError() << "controlBoards is specified, but it is not a list.";
+            return false;
+        }
+
+        yarp::os::Bottle* controlBoardsList = controlBoardsValue.asList();
+
+        for (size_t i = 0; i < controlBoardsList->size(); ++i)
+        {
+            yarp::os::Value cbValue = controlBoardsList->get(i);
+            if (!cbValue.isString())
+            {
+                yError() << "The value in position" << i << "(0-based) of controlBoards is not a string.";
+                return false;
+            }
+            m_controlBoards.push_back(cbValue.asString());
+        }
+    }
+
+    return true;
+}
+
+bool RemapperConnector::getJointNamesFromModel(const yarp::os::Searchable &inputConf, const iDynTree::Model &model)
+{
+    std::lock_guard<std::mutex> lock(m_basicInfo->mutex);
+
+    m_basicInfo->jointList.clear();
+
+    yarp::os::Value jointsValue = inputConf.find("joints");
+
+    if (jointsValue.isNull())
+    {
+        for (size_t i = 0; i < model.getNrOfJoints(); ++i)
+        {
+            if (model.getJoint(i)->getNrOfDOFs() == 1)
+            {
+                m_basicInfo->jointList.push_back(model.getJointName(i)); //automatically select all the joint in the model that have 1DOF
+            }
+        }
+        if (m_basicInfo->jointList.size() != model.getNrOfPosCoords())
+        {
+            yError() << "The model contains joints with more than 1DOF. This is not yet supported.";
+            return false;
+        }
```

Comment:
Maybe we could add a print here saying, automatically loading joint list from the model?

Reply:
Done in https://github.com/robotology/idyntree-yarp-tools/pull/7/commits/7166d81b4cb07058c45e7f8281012c9f39f3d42d